hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheService.java

```diff
@@ -157,12 +160,19 @@ public void commitMigration(PartitionMigrationEvent event) {
     public void rollbackMigration(PartitionMigrationEvent event) {
         if (event.getMigrationEndpoint() == MigrationEndpoint.DESTINATION) {
             clearPartitionReplica(event.getPartitionId());
+        } else {
+            // MigrationEndpoint.DESTINATION
+            resetPartitionReplica(event.getPartitionId(), false);
         }
     }
 
     @Override
     public void clearPartitionReplica(int partitionId) {
-        segments[partitionId].clear();
+        resetPartitionReplica(partitionId, true);
+    }
+
+    private void resetPartitionReplica(int partitionId, boolean clear) {
```

Comment:
I can remove `reset` (which contains clearing and refreshing cache record store) so cache service will call `clear` and `init` (or `refresh` or any other name. do you have any suggesstion for better name) sequentially after migration instead of a single `reset` call. 

@mdogan @ahmetmircik WDYT?